ui/pages/confirmations/confirmation/confirmation.js

```diff
@@ -210,9 +211,15 @@ export default function ConfirmationPage({
     useSafeChainsListValidationSelector,
   );
   const [approvalFlowLoadingText, setApprovalFlowLoadingText] = useState(null);
+
   const [currentPendingConfirmation, setCurrentPendingConfirmation] =
     useState(0);
-  const pendingConfirmation = pendingConfirmations[currentPendingConfirmation];
+  const pendingRoutedConfirmation = useConfirmationRouting();
+  // Confirmations that are directly routed to get priority and will be shown above the current queue.
+  const pendingConfirmation =
+    pendingRoutedConfirmation ??
```

Comment:
Just clarifying the intended behaviour is that if we use a bad ID, then it will fallback to the original queue as before rather than display an error?

Reply:
Yeah, that seems fine to me. But open to suggestions if you disagree.